min p1*x1^2 + p2*x2^2 + p3*x3^2

s.t. p1 >= 0, p2 >= 0, p3 >= 0, p1 + p2 + p3 = 5

#for simplicity, the p vector will be concatenated with the x vector 
#such that p123=x456

In [11]:
import numpy as np
from scipy.optimize import minimize 

In [12]:
fun = lambda x: (x[3]*x[0]**2) + (x[4]*x[1]**2) + (x[5]*x[2]**2)
cons = ({"type":"ineq","fun":lambda x: -x[0]+2*x[1]+4},
        {"type":"ineq","fun":lambda x: -x[0] - x[2] + x[1] + 2},
        {"type":"eq","fun":lambda x: x[3] + x[4] + x[5] - 5})
bnds = ((None,None),    #x1
        (None,None),    #x2
        (None,None),    #x3
        (0,None),       #p1
        (0,None),       #p2
        (0,None))       #p3

In [13]:
res = minimize(fun,np.zeros(6),constraints=cons,bounds=bnds)
res

     fun: 0.0
     jac: array([0., 0., 0., 0., 0., 0.])
 message: 'Optimization terminated successfully'
    nfev: 8
     nit: 1
    njev: 1
  status: 0
 success: True
       x: array([0.        , 0.        , 0.        , 1.66666667, 1.66666667,
       1.66666667])

Optimization Output:

     fun: 0.0
     jac: array([0., 0., 0., 0., 0., 0.])
 message: 'Optimization terminated successfully'
    nfev: 8
     nit: 1
    njev: 1
  status: 0
 success: True
       x: array([0.        , 0.        , 0.        , 1.66666667, 1.66666667,
       1.66666667])